src/index.jsx

```diff
@@ -48,7 +49,14 @@ const store = createStore(reducer, compose(
 store.dispatch(getMetadata());
 // Create an enhanced history that syncs navigation events with the store
 const history = syncHistoryWithStore(browserHistory, store);
-history.listen(location => store.dispatch(setActiveTab(location.pathname)));
+history.listen(({ state, pathname }) => {
+  if (state && state.tabChange) {
+    store.dispatch(setActiveTab(pathname));
+  } else if (pathname.includes('histograms')) {
+    // I'd like to do this better if there is a way
+    store.dispatch(setActiveTab(`${pathname.slice(0, pathname.lastIndexOf('/'))}/${histogramNames[0]}`));
```

Comment:
I think you should just have the selector/histogram components check this.props.routeParams and render themselves accordingly.